src/test/ij.c

```diff
@@ -2265,6 +2265,8 @@ main( hypre_int argc,
    else if ( build_matrix_type == 4 )
    {
       BuildParLaplacian27pt(argc, argv, build_matrix_arg_index, &parcsr_A);
+
+      hypre_CSRMatrixGpuSpMVAnalysis(hypre_ParCSRMatrixDiag(parcsr_A));
```

Comment:
Is this what we want eventually? But it's fine for me to leave it as is for now for testing. Thanks!

Reply:
This was what I had in mind and I put it here to serve as an example. I'm certainly open to suggestions/changes/etc.

Reply:
I should add it's what I had in mind as the low level, but user-facing API. Perhaps knobs could be added at the, e.g., `BoomerAMGSetup` level that will trigger calling this API on matrices used in the preconditioner.